src/main/java/org/example/StringCalculator.java

```diff
@@ -0,0 +1,64 @@
+package org.example;
+
+import java.util.ArrayList;
+import java.util.regex.Pattern;
+
+public class StringCalculator {
+    private int addCalledalledCount = 0;
+
+    public int getAddCalledalledCount() {
+        return addCalledalledCount;
+    }
+
+    public int getCalledCount() {
+        return addCalledalledCount;
+    }
+
+    public ArrayList<String> getDelimiterAndNumberString(String numbers) {
+        ArrayList<String> numberStringAndDelimiter = new ArrayList<>();
+        String numberString = numbers.substring(numbers.indexOf("]\n")+1).replace("\n","");
+        if (numbers.startsWith("//")) {
+            String delimiter = "";
+            String delimittedString = numbers.substring(numbers.indexOf("["), numbers.indexOf("\n"));
+
+            for (String delimit : delimittedString.replace("[","").split("]")) {
+                numberString = numberString.replaceAll(Pattern.quote(delimit), ",");
+            }
+
+            numberString = numberString.replaceAll("\n", "");
+            numberStringAndDelimiter.add(numberString);
+            numberStringAndDelimiter.add(delimiter);
+
+        } else {
+            numbers = numbers.replaceAll("\n", "");
+            numberStringAndDelimiter.add(numbers);
+            numberStringAndDelimiter.add(",");
+        }
+        return numberStringAndDelimiter;
+    }
+
+    public int add(String numbers) throws Exception {
```

Comment:
do not use throws use surround with try-catch use custom exception instead of generic one